ethclient/ethclient.go

```diff
@@ -265,7 +265,10 @@ func (ec *client) getBlock(ctx context.Context, method string, args ...interface
 		}
 		txs[i] = tx.tx
 	}
-	return types.NewBlockWithHeader(head).WithBody(txs, uncles).WithExtData(body.Version, (*[]byte)(body.BlockExtraData)), nil
+	block := types.NewBlockWithHeader(head)
+	block = block.WithBody(types.Body{Transactions: txs, Uncles: uncles})
+	block = types.WithBlockExtra(block, body.Version, (*[]byte)(body.BlockExtraData), false)
+	return block, nil
```

Comment:
we could have this into one line but it would be even more confusing imo, especially since it would start with `types.WithBlockExtras(types.NewBlockWithHeader(head), ...)`